pulp_python/app/tasks/publish.py

```diff
@@ -0,0 +1,167 @@
+  </body>
+</html>
+"""
+
+
+simple_detail_template = """<!DOCTYPE html>
+<html>
+<head>
+  <title>Links for {{ project_name }}</title>
+  <meta name="api-version" value="2" />
+</head>
+<body>
+    <h1>Links for {{ project_name }}</h1>
+    {% for name, path, md5 in project_packages %}
+    <a href="{{ path }}#md5={{ md5 }}" rel="internal">{{ name }}</a><br/>
+    {% endfor %}
+</body>
+</html>
+"""
+
+
+@shared_task(base=UserFacingTask)
+def publish(publisher_pk, repository_pk):
+    """
+    Use provided publisher to create a Publication based on a RepositoryVersion.
+    Args:
+        publisher_pk (str): Use the publish settings provided by this publisher.
+        repository_pk (str): Create a Publication from the latest version of this Repository.
+    """
+    publisher = python_models.PythonPublisher.objects.get(pk=publisher_pk)
+    repository = models.Repository.objects.get(pk=repository_pk)
+    latest_version = models.RepositoryVersion.latest(repository)
+
+    log.info(_('Publishing: repository={repo}, version={version}, publisher={pub}').format(
+        repo=repository.name,
+        version=latest_version.number,
+        pub=publisher.name
+    ))
+
+    with WorkingDirectory():
+        with models.Publication.create(latest_version, publisher) as publication:
+            write_simple_api(publication)
+
+    log.info(_('Publication: {pk} created').format(pk=publication.pk))
+
+
+def write_simple_api(publication):
+    """
+    Writes metadata mimicking the simple api of PyPI for all python packages in the
+    repository version
+
+    https://wiki.python.org/moin/PyPISimple
+
+    Args:
+        publication (pulpcore.plugin.models.Publication): A publication to generate metadata for
+    """
+
+    os.mkdir('simple')
+    project_names = python_models.PythonPackageContent.objects.order_by('name')\
+        .values_list('name', flat=True).distinct()
+
+    # generate a canonical name for each project name - the canonical name is all-lowercase
+    # with the names replaced by underscores
+    index_names = [(name, sanitize_name(name)) for name in project_names]
+
+    # write the root index, which lists all of the projects for which a package available
+    index_path = 'simple/index.html'
+    with open(index_path, 'w') as index:
+        context = Context({
+            'projects': index_names
+        })
+        template = Template(simple_index_template)
+        index.write(template.render(context))
+
+    index_md = models.PublishedMetadata(
+        relative_path=index_path,
+        publication=publication,
+        file=File(open(index_path, 'rb'))
+    )
+    index_md.save()
+
+    for (name, canonical_name) in index_names:
+        project_dir = 'simple/{}'.format(canonical_name)
+        os.mkdir(project_dir)
+
+        packages = python_models.PythonPackageContent.objects.filter(name=name)
+
+        package_detail_data = []
+
+        for package in packages:
+            filename = package.filename
```

Comment:
Optional style nitpick: Could you just use package.filename later? Seems unnecessary to make a new var.